.docs/test_plan.md

```diff
@@ -0,0 +1,190 @@
+
+-[Prerequisite and exigences](#prerequisite-and-exigences)
+
+-[The tools used to do the tests](#the-tools-used-to-do-the-tests)
+
+-[Tests environment](#Tests-environment)
+
+-[Exploited ressources](#exploited-ressources)
+
+-[The estimated results](#the-estimated-results)
+
+-[Tests to do](#tests-to-do)
+
+# Introduction
+
+This test plan describes the performance tests of ArmoniK. The purpose of those tests is to mesure the performance of the differents functionalities of ArmoniK (Tasks submission, Tasks treatment, ...). For this, we will do different tests:
+- Measure the execution time of different numbers of submitted tasks and different size of the tasks.
+- Test the strong scalability.
+
+The purpose of those tests is to have performance measurements in function of the ArmoniK versions. Since this is the first iteration of the performances tests we are going to establish our baseline in function of a specific versions (see after).
+
+# The product and the functionalities to test
+
+## Products
+
+### Stresstest "2.12.0"
+
+This test is a stress test which runs a number of tasks with a specific i/o sizes and duration of the tasks. It uses ArmoniK's c# extension. It copies a zip of the dlls of the application and install them on the workers.
+
+### Bench "0.11.4"
+
+This test is similar to StressTest but it doesn't use ArmoniK's c# extension, it means that the application is already in the docker image of the workers.
+
+## HtcMock "0.11.4" 
+
+With this test we can have an idea of how ArmoniK can handle the applications with subtasking and aggregation tasks.
+
+## functionalities
+
+- The submission of the tasks
+- The processing of the tasks
+- The retrieving of the results 
+
+# Prerequisite and exigences
+
+- Deploy ArmoniK with different partitions of bench (different number of pods in each partition)
+
+# The tools used to do the tests
+
+## Scripts bash
+
+- To start a warming up run then a batch of runs with the same parameters and store them in files
+
+## Python
+
+- Clean the files where we stored the data in readable json files.
+- Read the data and compute the median and the mean of each run batch.
+
+## Json 
+
+- We store the results of the performance tests in Json file which we will store in a database.
+
+# Tests environment
+
+- ArmoniK deployed on AWS via WSL2
+
+| Date | Infra version | Core version | API version  | Extension c# version | Instance type | Instance caracteristics | CPU frequency | nb vCPUs | RAM (GB) | Network bandwidth (Gbps) | EBS Bandwidth (Mbps) |
+| :---    | :---    | :---    | :---    | :---    | :---    | :---    | :---    | :---    | :---    | :---   |  :---   |
+| 03/29/2023 | 2.12.0 | 0.11.1 | 3.5.2 | 0.9.1 | c24.xlarge |  Intel Xeon de 2e (Cascade Lake) | 3.6 GHz - 3.9 GHz | 96 | 192 | 25 | 19000 |
+
+- [Config file](https://github.com/aneoconsulting/ArmoniK/blob/874-benchmarking/versions.tfvars.json) with all the versions of Armonik
```

Comment:
Be aware that you have to provide the link to versions.tfvars.json file of the correct ArmoniK version (here is ArmoniK v2.12.0).